pkg/yurttunnel/server/cmd.go

```diff
@@ -0,0 +1,182 @@
+	o := NewYurttunnelServerOptions()
+
+	cmd := &cobra.Command{
+		Use:   "yurttunnel-server",
+		Short: "yurttunnel-server sends requests to yurttunnel-agents",
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.validate(); err != nil {
+				return err
+			}
+			o.complete()
+			if err := o.run(stopCh); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&o.kubeConfig, "kube-config", o.kubeConfig,
+		"path to the kubeconfig file.")
+	flags.StringVar(&o.bindAddr, "bind-address", o.bindAddr,
+		"the ip address on which the server will listen.")
+	flags.StringVar(&o.caFile, "ca-file", o.caFile,
+		"CA file for connecting to edge tunnel agent and kubelet.")
+	flags.BoolVar(&o.enableIptables, "enable-iptables", o.enableIptables,
+		"if allow iptable manager to set the dnat rule.")
+	flags.IntVar(&o.iptablesSyncPeriod, "iptables-sync-period", o.iptablesSyncPeriod,
+		"the synchronization period of the iptable manager.")
+	return cmd
+}
+
+// YurttunnelServerOptions has the information that required by the
+// yurttunel-server
+type YurttunnelServerOptions struct {
+	kubeConfig               string
+	bindAddr                 string
+	caFile                   string
+	enableIptables           bool
+	iptablesSyncPeriod       int
+	serverAgentPort          int
+	serverMasterPort         int
+	interceptorServerUDSFile string
+	serverAgentAddr          string
+	serverMasterAddr         string
+}
+
+// NewYurttunnelServerOptions creates a new YurtNewYurttunnelServerOptions
+func NewYurttunnelServerOptions() *YurttunnelServerOptions {
+	o := &YurttunnelServerOptions{
+		caFile:                   "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
+		bindAddr:                 "0.0.0.0",
+		enableIptables:           true,
+		iptablesSyncPeriod:       60,
+		serverAgentPort:          constants.YurttunnelServerAgentPort,
+		serverMasterPort:         constants.YurttunnelServerMasterPort,
+		interceptorServerUDSFile: "/tmp/interceptor-proxier.sock",
+	}
+	return o
+}
+
+// validate validates the YurttunnelServerOptions
+func (o *YurttunnelServerOptions) validate() error {
+	if len(o.bindAddr) == 0 {
+		return fmt.Errorf("tunnel server's bind address can't be empty")
+	}
+	return nil
+}
+
+// complete completes all the required options
+func (o *YurttunnelServerOptions) complete() {
+	o.serverAgentAddr = fmt.Sprintf("%s:%d", o.bindAddr, o.serverAgentPort)
+	o.serverMasterAddr = fmt.Sprintf("%s:%d", o.bindAddr, o.serverMasterPort)
+	klog.Infof("server will accept agent requests at: %s, "+
+		"server will accept master requests at: %s",
+		o.serverAgentAddr, o.serverMasterAddr)
+}
+
+// run starts the yurttunel-server
+func (o *YurttunnelServerOptions) run(stopCh <-chan struct{}) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	// 1. create the master's clientset
+	clientSet, csErr := kubernetes.CreateClientSet(o.kubeConfig)
```

Comment:
kubeConfig may not exist, could we add another parameter as a backup?

Reply:
`kubeconfig` will be a mandatory option.